src/c_api/c_api_function.cc

```diff
@@ -56,7 +56,7 @@ std::vector<nnvm::NodeEntry> Gradient(
 
   std::vector<nnvm::NodeEntry> ret;
   for (index_t i = 0; i < g->num_outputs(); ++i) {
-    ret.emplace_back(nnvm::NodeEntry{g, i, 0});
+    ret.emplace_back(nnvm::NodeEntry{g, static_cast<uint32_t>(i), 0});
```

Comment:
For large values of `i` this cast would fail, right? Probably another change in nnvm is needed to use `index_t` for the index: [node.h](https://github.com/dmlc/nnvm/blob/ef0ab9b09dbf1318851be311d3752de6c9bd4881/include/nnvm/node.h#L35)

Reply:
The Node data structure in nnvm::Graph uses uint32_t to index graph nodes. In that case, the number of nodes in the graph is not going to be greater than 2^32. I have communicated this to the tvm team and they suggest we keep the uint32_t for node id.

Reply:
@apeforest Ok, but then I think it would make sense to indicate an error here somehow, because if it doesn't fit the behaviour is undefined.